server/runner/lib/runner/qa/tools.ex

```diff
@@ -738,22 +738,18 @@ defmodule Runner.QA.Tools do
         })
       ],
       function: fn %{"step_id" => step_id, "result" => result, "issues" => issues} = _params, _context ->
-        case Client.update_step(%{
-               step_id: step_id,
-               result: result,
-               issues: issues,
-               server_url: server_url,
-               run_id: run_id,
-               auth_token: auth_token,
-               account_handle: account_handle,
-               project_handle: project_handle
-             }) do
-          {:ok, _} ->
-            {:ok, "Step report submitted successfully."}
+        Client.update_step(%{
```

Comment:
I'd match this one against `{:ok, :async}` to ensure there's a contract match and let the type system warn us if we are missing return types.